src/txn/txn_timestamp.c

```diff
@@ -553,14 +567,28 @@ __wt_timestamp_validate(WT_SESSION_IMPL *session, const char *name,
 	    txn_global->has_stable_timestamp &&
 	    __wt_timestamp_cmp(ts, &txn_global->stable_timestamp) < 0));
 
-	if (older_than_oldest_ts)
+	if (older_than_oldest_ts) {
+		/*
+		 * As global oldest timestamp access is for error message only,
```

Comment:
I think you should acquire a lock in this case - it's not in a performance path so doesn't warrant an exception, and without a lock we can see invalid timestamps (i.e: a timestamp might be half way through being updated). It would actually be nice if this value was identical to the one used in the comparison (i.e: we got a local copy) otherwise the error message could be confusing, since the state may have changed.